packages/esm-patient-vitals-app/src/vitals/print-this.component.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useRef } from 'react';
+import { Printer } from '@carbon/react/icons';
+import { useReactToPrint } from 'react-to-print';
+import { useTranslation } from 'react-i18next';
+import styles from './print-this.scss';
+import { useSession } from '@openmrs/esm-framework';
+import { Header, Button } from '@carbon/react';
+
+interface Props {
+  subheader?: any;
+  body: any;
+  footer?: any;
+  requestedBy: string;
+  patientDetails?: any;
+}
+
+export class PrintComponent extends React.Component<Props, {}> {
+  render() {
+    const { subheader, body, patientDetails, requestedBy } = this.props;
+
+    return (
+      <div className={`${styles.yourClassName} `}>
+        <Header>
+          <div className={styles.spacedLogo}>
+            <img
+              src="https://labsmart-healthcare-trial.s3.us-west-2.amazonaws.com/diagnostic_lab_524/sample_report_header.jpg"
+              alt="header"
+              width={110}
+              height={60}
+            />
+          </div>
+        </Header>
+        <div className={`${styles.reportcontent} `}>
+          <div className={`${styles.reportheader} `}></div>
+
+          <div className={`${styles.patientdetails} `}>
+            <p>{patientDetails?.name}</p>
+            <p>{patientDetails?.age}</p>
+            <p> {patientDetails?.gender}</p>
+            <p>{patientDetails?.location}</p>
+          </div>
+          <hr></hr>
+          <div className={`${styles.subheader} `}>
+            <h1> {subheader} </h1>
+          </div>
+          <hr></hr>
+          {body}
+          <div className={`${styles.footer} `}>Requested by {requestedBy}</div>
+        </div>
+      </div>
+    );
+  }
+}
+
+const PrintThisComponent = ({ subheader, body, patientDetails }) => {
+  const componentRef = useRef();
+  const { t } = useTranslation();
+  const handlePrint = useReactToPrint({
+    content: () => componentRef.current,
+  });
+  const session = useSession();
+  return (
+    <div>
+      <Button kind="ghost" renderIcon={Printer} iconDescription="Add vitals" onClick={handlePrint}>
+        {t('print', 'Print')}
```

Comment:
This should probably be just more than `print`, something like `print vitals`. This is because you will have many of these in many places.

Reply:
Actually, looking at where the button is, just print might be okay since it is inside the Vitals table